server/src/main/java/org/elasticsearch/cluster/metadata/ReservedStateMetadata.java

```diff
@@ -214,6 +214,20 @@ public Builder(String namespace) {
             this.errorMetadata = null;
         }
 
+        /**
+         * Creates an reserved state metadata builder
+         *
+         * @param metadata  the previous metadata
+         */
+        public Builder(ReservedStateMetadata metadata) {
+            this(metadata.namespace);
+            if (metadata != null) {
```

Comment:
Is this null check redundant? On the previous line, `metadata.namespace` will throw a NPE if metadata is null